### ISSUE TITLE
Automatic fallback to llvmpipe

### ISSUE THREAD

**Issue by jessstrap** [servo/servo#13653]
If the systems OpenGL driver doesn't fulfil the needs of webrender fall back to using llvmpipe/osmesa instead of panicking. 

See these [log](http://logs.glob.uno/?c=mozilla%23servo&s=8+Oct+2016&e=8+Oct+2016#c538516) [entries](http://logs.glob.uno/?c=mozilla%23servo&s=8+Oct+2016&e=8+Oct+2016#c538552) for a starting place. Getting this to work on linux and mac should be pretty easy. Getting this to work on windows will probably be more difficult.

**Comment by jessstrap** [comment]
I'd like to take this one. 

It may also be a good candidate for an E-Easy tag.

**Comment by larsbergstrom** [comment]
Thanks for doing this! It will be quite helpful not just for old drivers but also for some virtual machines (e.g., VirtualBox) whose graphics drivers have historically not supported the full set of features we need for WR via normal OpenGL.

**Comment by nox** [comment]
Was this done?

**Comment by jdm** [comment]
No, hence why there are issues about being unable to create a GL context.

**Comment by jessstrap** [comment]
I have not made any progress on this issue in the last.... year. :man_facepalming: If anyone else wants to work it, have at. (I'm not sure if it was ever officially assigned to me in the first place.)

**Comment by nox** [comment]
@jessstrap Ok thanks for the follow-up anyway!

**Comment by SimonSapin** [comment]
Is this still relevant after https://github.com/servo/servo/pull/25853?

**Comment by jdm** [comment]
Yes, this is separate from whether we use osmesa for headless mode or not.
